src/chainparams.cpp

```diff
@@ -118,7 +118,11 @@ class CMainParams : public CChainParams {
 
         genesis = CreateGenesisBlock(1231006505, 2083236893, 0x1d00ffff, 1, 50 * COIN);
         consensus.hashGenesisBlock = genesis.GetHash();
-        assert(consensus.hashGenesisBlock == uint256S("0x000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f"));
+        consensus.blockheight_in_header = gArgs.GetBoolArg("-con_blockheightinheader", false);
```

Comment:
No, just don't use -con_blockheightinheader at all outside of CCustomParams and you shouldn't need to touch neither CMainParams, CTestNetParams or CRegTestParams, specially not their asserts.
Perhaps it would be interesting to rebase on top of https://github.com/ElementsProject/elements/pull/436 and add a test like this:

    {
        'memo': 'default_style with height in header',
        'genesis': 'c03f16ae9e2980de2b61fd6dc84af8ac4a37bea928af632166a6b36c5c871ddd',
        'args': [
                '-con_genesis_style=default_style',
                '-con_blockheightinheader',
        ],
    },

Reply:
It creates a new style genesis block for each network when iterating through network types during load, then hits the assert.

Reply:
That means blockheight_in_header is not really acting as false for main, test and regtest, no?

Reply:
It is, if you don't set `-con_blockheightinheader` to true